src/metabolic_ninja/tasks.py

```diff
@@ -333,17 +338,57 @@ def evaluate_cofactor_swap(designs, pathway, model, method):
     if designs is None:
         return []
     logger.info(f"Evaluating {len(designs)} co-factor swap designs.")
+    source_pair = ("nad_c", "nadh_c")
+    target_pair = ("nadp_c", "nadph_c")
     results = []
-    for row in designs.data_frame.itertuples(index=False):
+    # FIXME (Moritz Beber): The model context is currently bugged.
+    #  See https://github.com/opencobra/cobrapy/issues/849
+    #  We need to make a copy.
+    model = model.copy()
+    source_a = model.metabolites.get_by_id(source_pair[0])
+    source_b = model.metabolites.get_by_id(source_pair[1])
+    target_a = model.metabolites.get_by_id(target_pair[0])
+    target_b = model.metabolites.get_by_id(target_pair[1])
+    for design in designs.data_frame.itertuples(index=False):
+        manipulations = []
+        for rxn_id in design.targets:
+            rxn: Reaction = model.reactions.get_by_id(rxn_id)
+            metabolites = rxn.metabolites
+            # Swap from source to target co-factors.
+            if source_a in metabolites:
+                metabolites[target_a] = metabolites[source_a]
+                metabolites[target_b] = metabolites[source_b]
+                metabolites[source_a] = 0
+                metabolites[source_b] = 0
+                manipulations.append({"id": rxn_id, "from": [source_pair[0], source_pair[1]], "to": [target_pair[0], target_pair[1]]})
```

Comment:
I think the JSON serializer will automatically convert a tuple into a list so you simply use:
```suggestion
                manipulations.append({"id": rxn_id, "from": source_pair, "to": target_pair})
```